pkg/crc/machine/types.go

```diff
@@ -24,7 +25,7 @@ type StartConfig struct {
 	Debug bool
 
 	// User Pull secret
-	GetPullSecret GetPullSecretFunc
+	PullSecret *cluster.PullSecret
```

Comment:
> I don't see much difference in behaviour with or without this type, before we had a separate callback and string, now we have a struct holding the callback and the string, but ultimately we call the callback at about the same time, and we the struct will be carrying a string all along. In a way, the new type is somehow worse than the explicit start.GetPullSecret(), it's not obvious that _, err = startConfig.PullSecret.Value() is here to ensure we ask for the pull secret.
I don't mind this change, but I feel you are overselling it ;)

Yes you're right but now, with the EnsurePullSecret* functions, if by any chance, the pull secret is required it will need to ask for it. We don't know in advance if the pull secret will be needed or not in this case.